### ISSUE TITLE
Test for WCAG success criterion 4.1.1: Parsing

### ISSUE THREAD

**Issue by hannolans** [quailjs/quail#163]
http://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-parses.html
In content implemented using markup languages, elements have complete start and end tags, elements are nested according to their specifications, elements do not contain duplicate attributes, and any IDs are unique, except where the specifications allow these features.
## Scope selector

Source HTML of a page
## Applicability

This succes criterion is always applicable.
## HTML Failures
- [x] F70: Incorrect use of start and end tags or attribute markup (#161)
- [x] F77: Duplicate values of type ID (#164 - implemented)
- [x] F17: Insufficient information in DOM to determine one-to-one relationships (#162)

Note: only HTML failures are mentioned
## Sufficient HTML Techniques
- [ ] G134: Validating Web pages (#165)
- [ ] G192: Fully conforming to specifications (#89)
- [ ] H88: Using HTML according to spec (#86)
- [x] H74: Ensuring that opening and closing tags are used according to specification (#88 - implemented) AND H93: Ensuring that id attributes are unique on a page (#167 - implemented)  AND H94: Ensuring that elements do not contain duplicate attributes (#67 implemented)
## Relationship between Failures and Techniques

There is a certain dependency between these techniques and failures. We should define that as possibly we can use one common library that can give us the outcomes.
- G134 passes when a validation tool accepts the HTML
- H88 passes when a validation tool accepts the HTML (G134) AND the HTML is semantic (using techniques as in SC 1.3.1 - Info and Relationships: H51 + H48 + H42 + ...)
- H88 seems the HTML specific version for G192, so when H88 passes, G192 passes.
- H74+H93+H94 is a lighter variant of G134, as this allows elements that are not in the specification.
- if G134 OR G192 OR H88 passes, H74, H93 and H94 pass
- If F17 fails, F77 fails as well, as duplicate ids (F77) is one of the tests of F17
- H93 and F77 are counterpart tests: if H93 passes, F77 passes as well, if F77 fails, H93 fails as well.

Note that F70 seems more strict than HTML5 for "Failure to provide a closing element tag for elements that do not accept empty-element syntax.", but in HTML5 for example a p element doesn't need a closing tag anymore. So even if a page validates one of the above techniques, it can still fail for F70.
